### DISCUSSION TITLE
Error while building

### DISCUSSION THREAD

**Discussion by slch** [tauri-apps/tauri#3269] [Q&A] [answered]
Hi!

I'm new to tauri.
After slightly modifying js src, got the following error during build (while creating AppDir?)

```
...
[gtk/stdout] Deploying shared library /usr/lib/vmware/lib/libgobject-2.0.so.0/libgobject-2.0.so.0
[gtk/stdout] Deploying dependencies for ELF file /usr/lib/vmware/lib/libgobject-2.0.so.0/libgobject-2.0.so.0 
[gtk/stdout] ERROR: Could not find dependency: libffi.so.6 
[gtk/stdout] ERROR: Failed to deploy library:  /usr/lib/vmware/lib/libgobject-2.0.so.0/libgobject-2.0.so.0 
ERROR: Failed to run plugin: gtk (exit code: 1) 
```

Here is the full output of `yarn tauri build --verbose`: [out.txt](https://github.com/tauri-apps/tauri/files/7921251/out.txt)
`yarn tauri dev` works fine.

<details>
<summary>
yarn tauri info
</summary>

yarn run v1.22.17
$ tauri info
 app:spawn Running "/home/slch/d/js/hello-tauri/node_modules/@tauri-apps/cli/bin/tauri-cli tauri info" +0ms

Operating System - Arch Linux, version Rolling Release X64

Node.js environment
  Node.js - 17.3.0
  @tauri-apps/cli - 1.0.0-beta.10
  @tauri-apps/api - 1.0.0-beta.8

Global packages
  npm - 8.3.2
  yarn - 1.22.17

Rust environment
  rustc - 1.58.0
  cargo - 1.58.0

App directory structure
/node_modules
/src-tauri
/build
/src
/public

App
  tauri.rs - 1.0.0-beta.8
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - ../build
  devPath - http://localhost:3000
  framework - React

Done in 4.54s.

</details>

<details>
<summary>
Using arch btw
</summary>

Operating System: Arch Linux

KDE Plasma Version: 5.23.5

KDE Frameworks Version: 5.90.0

Qt Version: 5.15.2

Kernel Version: 5.16.2-arch1-1 (64-bit)

Graphics Platform: X11
</details>

**Comment by nothingismagick** [comment]
Please edit your question with the results of `yarn tauri info`. Thankyou!

**Reply by slch** [reply]
Updated!

**Comment by Testare** [comment]
Same issue.

**Comment by unitythemaker** [comment, marked as answer]
I got the same issue in Arch Linux. Fixed with ``paru -S libffi6``

**Reply by slch** [reply]
Worked for me (with `yay`).
Should have tried installing the package :smile_cat:

**Comment by slch** [comment]
I expected being able to build for all platforms (linux, windows, mac) on single (linux) platform... Which isn't the case sadly.
For now only dev locally and all building should be done via github actions or analogs.

**Comment by Raduc4** [comment]
Has anyone solved this error on Arch Linux?
`ERROR: Could not find dependency: libicui18n.so.72`
```
pacman -Qi icu
Name            : icu
Version         : 73.2-2
```

**Reply by vjousse** [reply]
Yes, running `yay -S icu72` fixes the problem.

**Comment by AshutoshSajan** [comment]
i am getting similar kind of error

<details>
  <summary>pnpm tauri info</summary>
  
> dorion@4.1.2 tauri /home/ashu/Documents/Open Source/Dorion
> tauri "info"


[✔] Environment
    - OS: Ubuntu 20.04 X64
    ✔ webkit2gtk-4.0: 2.38.6
    ✔ rsvg2: 2.48.9
    ✔ rustc: 1.76.0 (07dca489a 2024-02-04)
    ✔ cargo: 1.76.0 (c84b36747 2024-01-18)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-unknown-linux-gnu (default)
    - node: 18.18.2
    - pnpm: 8.9.2
    - yarn: 1.22.19
    - npm: 9.8.1
    - bun: 1.0.21

[-] Packages
    - tauri [RUST]: 1.5.4
    - tauri-build [RUST]: 1.5.1
    - wry [RUST]: 0.24.7
    - tao [RUST]: 0.16.7
    - tauri-cli [RUST]: 1.5.9
    - @tauri-apps/api [NPM]: 1.5.1 (outdated, latest: 1.5.3)
    - @tauri-apps/cli [NPM]: 1.5.6 (outdated, latest: 1.5.9)

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: https://discord.com/app
    - devPath: https://discord.com/app
    - bundler: Vite

</details>

> system info

![image](https://github.com/tauri-apps/tauri/assets/44765540/f7bb2819-2b7b-4ac4-b338-a777d6348ffd)

> error while building

```console
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/ashu/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/ashu/Documents/Open Source/Dorion/

```

```console
 = note: /usr/bin/ld: /home/ashu/Documents/Open Source/Dorion/src-tauri/target/debug/deps/libopenssl-218892c00800079b.rlib(openssl-218892c00800079b.openssl.61cde6a6cf2e7a55-cgu.01.rcgu.o): in function `openssl::ssl::SslRef::peer_certificate':
          /home/ashu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openssl-0.10.63/src/ssl/mod.rs:2550: undefined reference to `SSL_get1_peer_certificate'
          /usr/bin/ld: /home/ashu/Documents/Open Source/Dorion/src-tauri/target/debug/deps/libopenssl-218892c00800079b.rlib(openssl-218892c00800079b.openssl.61cde6a6cf2e7a55-cgu.10.rcgu.o): in function `openssl::error::Error::get':
          /home/ashu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openssl-0.10.63/src/error.rs:122: undefined reference to `ERR_get_error_all'
          collect2: error: ld returned 1 exit status
          
  = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)
  ```

> when installed from deb package, got this error.

```console
dorion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by dorion)
dorion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by dorion)
dorion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by dorion)
```
